services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/subscription/SubscriptionManagerImpl.java

```diff
@@ -118,4 +124,34 @@ public Etag getEtag(String objectId, ObjectType objectType) {
 		etag.setEtag(changeDao.getEtag(objectIdLong, objectType));
 		return etag;
 	}
+
+	@Override
+	public SubscriberPagedResults getSubscribers(UserInfo userInfo, Topic topic, String nextPageToken) {
+		ValidateArgument.required(userInfo, "userInfo");
+		ValidateArgument.required(topic, "topic");
+		ValidateArgument.required(topic.getObjectId(), "Topic.objectId");
+		ValidateArgument.required(topic.getObjectType(), "Topic.objectType");
+		AuthorizationManagerUtil.checkAuthorizationAndThrowException(
+				authorizationManager.canSubscribe(userInfo, topic.getObjectId(), topic.getObjectType()));
+		NextPageToken token = null;
+		if(nextPageToken != null){
+			token = new NextPageToken(nextPageToken);
+		}else{
+			token = new NextPageToken(DEFAULT_LIMIT, DEFAULT_OFFSET);
+		}
+		if(token.getLimit() > MAX_LIMIT){
+			throw new IllegalArgumentException("Limit must not exceed: "+MAX_LIMIT);
+		}
+		List<String> subscribers = subscriptionDao.getSubscribers(topic.getObjectId(), topic.getObjectType(), token.getLimit()+1, token.getOffset());
+		SubscriberPagedResults results = new SubscriberPagedResults();
+		if(subscribers.size() > token.getLimit()){
```

Comment:
add method string <T> getNextPageTokenForCurrentResult(List<T> results):
* check for size
* construct the next page token
* remove the last value from the list
* document about modified the given list

results.setNextPageToken(token.getNextPageTokenForCurrentResults(subscribers));